src/test/java/com/softensy/SoftensySpringBoot/controller/DoctorControllerTest.java

```diff
@@ -0,0 +1,119 @@
+                .andExpect(jsonPath("$.id").value("1"))
+                .andExpect(jsonPath("$.firstName").value("Ivan"))
+                .andExpect(jsonPath("$.lastName").value("Ivanov"))
+                .andExpect(jsonPath("$.middleName").value("Ivanovich"))
+                .andExpect(jsonPath("$.position").value("Hirurg"))
+                .andExpect(jsonPath("$.dateOfBirth").value("11-05-1987"))
+                .andExpect(jsonPath("$.phoneNamber").value("12345"));
+    }
+
+
+    @Test
+    void givenDoctor_whenAdd_thenStatus201() throws Exception {
+        //given
+        Doctor firstDoctor = new Doctor("Ivan", "Ivanov", "Ivanovich", "Hirurg", Date.valueOf("1987-05-12"), 12345L);
+        firstDoctor.setId(1);
+        //when
+        when(doctorService.saveDoctor(firstDoctor)).thenReturn(firstDoctor);
+        //then
+        mockMvc.perform(post("/doctor")
+                        .content(objectMapper.writeValueAsString(firstDoctor))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    void givenDoctor_whenUpdate_thenStatus200() throws Exception {
+        //given
+        Doctor firstDoctor = new Doctor("Ivan", "Ivanov", "Ivanovich", "Hirurg", Date.valueOf("1987-05-12"), 12345L);
+        firstDoctor.setId(1);
+        //when
+        when(doctorService.updateDoctor(firstDoctor)).thenReturn(firstDoctor);
+        //then
+        mockMvc.perform(put("/doctor")
+                        .content(objectMapper.writeValueAsString(firstDoctor))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void givenDoctor_whenDeleteDoctor_thenStatus204() throws Exception {
+        //given
+        Doctor firstDoctor = new Doctor("Ivan", "Ivanov", "Ivanovich", "Hirurg", Date.valueOf("1987-05-12"), 12345L);
+        firstDoctor.setId(1);
+        //when
+        when(doctorService.getDoctorById(1l)).thenReturn(Optional.of(firstDoctor));
+        //then
+        mockMvc.perform(delete("/doctor/1"))
+                .andExpect(status().isNoContent());
+    }
+
+    @Test
+    void givenDoctors_whenGetDoctors_thenStatus200() throws Exception {
+        //given
+        List<Doctor> doctors = new ArrayList<>();
+        Doctor firstDoctor = new Doctor("Ivan", "Ivanov", "Ivanovich", "Hirurg", Date.valueOf("1987-05-12"), 12345L);
+        firstDoctor.setId(1);
+        Doctor secondDoctor = new Doctor("Petr", "Petrov", "Petrov", "Terapevt", Date.valueOf("1988-07-19"), 54321L);
+        secondDoctor.setId(2);
+        doctors.add(firstDoctor);
+        doctors.add(secondDoctor);
+        //when
+        when(doctorService.getAllDoctors()).thenReturn(doctors);
+        //then
+        mockMvc.perform(get("/doctor"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(objectMapper.writeValueAsString(Arrays.asList(firstDoctor, secondDoctor))));
```

Comment:
you should have ready String representation of web response, not received from some code